File: backend/src/domain/entities/Owner.js
// The Owner entity represents the shop manager who has full access to the application.
class Owner {
    constructor({ 
        id, 
        name, 
        shopName, // The name of the business entity
        phone, 
        email, 
        password, // Encrypted password string
        createdAt, 
        updatedAt,
        role,
        status,
        isSuspended
    }) {
        this.id = id;
        this.name = name || '';
        this.shopName = shopName || '';
        this.phone = phone || '';
        this.email = email || '';
        this.password = password || '';
        this.status = status || 'approved';
        this.isSuspended = isSuspended ?? false;
        this.createdAt = createdAt || new Date().toISOString();
        this.updatedAt = updatedAt || new Date().toISOString();
        this.role = role || 'owner';
    }

    toJSON() {
        return {
            id: this.id,
            name: this.name,
            shopName: this.shopName,
            phone: this.phone,
            email: this.email,
            status: this.status,
            isSuspended: this.isSuspended,
            createdAt: this.createdAt,
            updatedAt: this.updatedAt,
            role: this.role
        };
    }
}

module.exports = Owner;

File: frontend/lib/features/auth/domain/entities/owner.dart
// The Owner entity represents a shop manager's account details in the Flutter app.
class Owner {
  final String id;
  final String name;
  final String shopName;
  final String phone;
  final String email;
  final String status;
  final bool isSuspended;
  final String createdAt;

  Owner({
    required this.id,
    required this.name,
    this.shopName = '', // The name of the business they manage
    this.phone = '',
    required this.email,
    this.status = 'approved',
    this.isSuspended = false,
    this.createdAt = '',
  });

  factory Owner.fromJson(Map<String, dynamic> json) {
    return Owner(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      shopName: json['shopName'] ?? '',
      phone: json['phone'] ?? '',
      email: json['email'] ?? '',
      status: json['status'] ?? 'approved',
      isSuspended: json['isSuspended'] ?? false,
      createdAt: json['createdAt'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'shopName': shopName,
      'phone': phone,
      'email': email,
      'status': status,
      'isSuspended': isSuspended,
    };
  }
}
